crates/rspack_plugin_javascript/src/plugin.rs

```diff
@@ -573,3 +576,60 @@ impl Plugin for JsPlugin {
     Ok(())
   }
 }
+
+#[derive(Debug)]
+pub struct InferAsyncModulesPlugin;
+
+#[async_trait::async_trait]
+impl Plugin for InferAsyncModulesPlugin {
+  fn name(&self) -> &'static str {
+    "InferAsyncModulesPlugin"
+  }
+
+  fn module_ids(&mut self, compilation: &mut Compilation) -> Result<()> {
```

Comment:
This hook should be `finishModules`, you can add `finishModules` on `Plugin` trait in this PR or a following PR.